### PR TITLE
M10 fix: allow submitting the signatures on any order

### PR DESCRIPTION
**Pull request by fedealconada**
Fixes M10 bug from Pessimistic audit

P.S test for this is on the `increase-coverage-2` branch which is a WIP PR.

### CODE REVIEW

File: src/wallet/KintoWallet.sol [hunk 1]
```diff
@@ -286,22 +287,28 @@ contract KintoWallet is Initializable, BaseAccount, TokenCallbackHandler, IKinto
...
+                if (owners[j] == recovered && !hasSigned[j]) {
+                    hasSigned[j] = true;
+                    requiredSigners--;
+                    break; // once the owner is found
                 }
             }
         }
-        return _packValidationData(requiredSigners != 0, 0, 0);
+
+        if (requiredSigners == 0) {
```
Comment (rrecuero) [inline src/wallet/KintoWallet.sol:307]: Why not keep this? Its smaller and equivalent

return _packValidationData(requiredSigners != 0, 0, 0);

**Review by rrecuero — CHANGES_REQUESTED** [review]
Looks good to merge after the comment

File: src/wallet/KintoWallet.sol (hunk 1 above)
Comment (fedealconada) [reply src/wallet/KintoWallet.sol:307]: yep, i was testing something and left it like that, but agree it is much cleaner the other way! changing it now

Comment (fedealconada) [reply src/wallet/KintoWallet.sol]: pushed!

**Pull request merged**